src/coreclr/pal/src/locale/unicode.cpp

```diff
@@ -253,16 +253,11 @@ MultiByteToWideChar(
         goto EXIT;
     }
 
-    // Use UTF8ToUnicode on all systems, since it replaces
+    // Use minipal_utf8_to_utf16_preallocated on all systems, since it replaces
     // invalid characters and Core Foundation doesn't do that.
     if (CodePage == CP_UTF8 || CodePage == CP_ACP)
     {
-        if (cbMultiByte <= -1)
-        {
-        cbMultiByte = strlen(lpMultiByteStr) + 1;
-        }
-
-        retval = UTF8ToUnicode(lpMultiByteStr, cbMultiByte, lpWideCharStr, cchWideChar, dwFlags);
+        retval = minipal_utf8_to_utf16_preallocated(lpMultiByteStr, cbMultiByte, (CHAR16_T**)&lpWideCharStr, cchWideChar, dwFlags);
```

Comment:
`minipal_utf8_to_utf16_preallocated` return `size_t`. `MultiByteToWideChar` returns `int`. There is a potential for silent overflow. 

I would have expected that the compiler will warn about this.

Reply:
Clang warns it with `-Wshorten-64-to-32` when it is explicitly specified, which we don't have.